src/titiler/xarray/titiler/xarray/io.py

```diff
@@ -0,0 +1,258 @@
+                if x in da.dims
+            )
+        except StopIteration as e:
+            raise ValueError(f"Couldn't find X/Y dimensions in {da.dims}") from e
+
+        da = da.rename({latitude_var_name: "y", longitude_var_name: "x"})
+
+    if "TIME" in da.dims:
+        da = da.rename({"TIME": "time"})
+
+    if _dims := [d for d in da.dims if d not in ["x", "y"]]:
+        da = da.transpose(*_dims, "y", "x")
+    else:
+        da = da.transpose("y", "x")
+
+    # If min/max values are stored in `valid_range` we add them in `valid_min/valid_max`
+    vmin, vmax = da.attrs.get("valid_min"), da.attrs.get("valid_max")
+    if "valid_range" in da.attrs and not (vmin is not None and vmax is not None):
+        valid_range = da.attrs.get("valid_range")
+        da.attrs.update({"valid_min": valid_range[0], "valid_max": valid_range[1]})
+
+    return da
+
+
+def get_variable(
+    ds: xarray.Dataset,
+    variable: str,
+    datetime: Optional[str] = None,
+    drop_dim: Optional[str] = None,
+) -> xarray.DataArray:
+    """Get Xarray variable as DataArray.
+
+    Args:
+        ds (xarray.Dataset): Xarray Dataset.
+        variable (str): Variable to extract from the Dataset.
+        datetime (str, optional): datetime to select from the DataArray.
+        drop_dim (str, optional): DataArray dimension to drop in form of `{dimension}={value}`.
+
+    Returns:
+        xarray.DataArray: 2D or 3D DataArray.
+
+    """
+    da = ds[variable]
+
+    if drop_dim:
+        dim_to_drop, dim_val = drop_dim.split("=")
+        da = da.sel({dim_to_drop: dim_val}).drop_vars(dim_to_drop)
+
+    da = _arrange_dims(da)
+
+    # Make sure we have a valid CRS
+    crs = da.rio.crs or "epsg:4326"
+    da = da.rio.write_crs(crs)
+
+    if crs == "epsg:4326" and (da.x > 180).any():
+        # Adjust the longitude coordinates to the -180 to 180 range
+        da = da.assign_coords(x=(da.x + 180) % 360 - 180)
+
+        # Sort the dataset by the updated longitude coordinates
+        da = da.sortby(da.x)
+
+    # TODO: Technically we don't have to select the first time, rio-tiler should handle 3D dataset
+    if "time" in da.dims:
+        if datetime:
+            # TODO: handle time interval
+            time_as_str = datetime.split("T")[0]
+            if da["time"].dtype == "O":
+                da["time"] = da["time"].astype("datetime64[ns]")
+
+            da = da.sel(
+                time=numpy.array(time_as_str, dtype=numpy.datetime64), method="nearest"
+            )
+        else:
+            da = da.isel(time=0)
+
+    assert len(da.dims) in [2, 3], "titiler.xarray can only work with 2D or 3D dataset"
+
+    return da
+
+
+@attr.s
+class Reader(XarrayReader):
+    """Reader: Open Zarr file and access DataArray."""
+
+    src_path: str = attr.ib()
+    variable: str = attr.ib()
+
+    # xarray.Dataset options
+    opener: Callable[..., xarray.Dataset] = attr.ib(default=xarray_open_dataset)
```

Comment:
For now the `opener` MUST be a callable that take 4 arguments:
- src_path: str
- group: Any
- decode_times: bool 
- cache_client 

we might change this in the future